composer/loggers/progress_bar_logger.py

```diff
@@ -105,140 +128,178 @@ def __init__(
         console_log_level: Union[LogLevel, str, Callable[[State, LogLevel], bool]] = LogLevel.EPOCH,
         stream: Union[str, TextIO] = sys.stderr,
     ) -> None:
-        self.show_pbar = progress_bar
-        log_to_console = not progress_bar if log_to_console is None else log_to_console
-        if not log_to_console:
-            console_log_level = lambda state, ll: False
-        self.pbars: Dict[bool, _ProgressBarLoggerInstance] = {}
+
+        self._show_pbar = progress_bar
+        self.train_pbar: Optional[_ProgressBar] = None
+        self.eval_pbar: Optional[_ProgressBar] = None
         self.is_train: Optional[bool] = None
 
         if isinstance(console_log_level, str):
             console_log_level = LogLevel(console_log_level)
-        if isinstance(console_log_level, LogLevel):
 
-            def should_log(state: State, log_level: LogLevel, console_log_level: LogLevel = console_log_level):
-                del state  # unused
-                return log_level <= console_log_level
+        if log_to_console is None:
+            log_to_console = not progress_bar
 
-            self.should_log = should_log
+        if not log_to_console:
+            # never log to console
+            self.should_log = lambda state, ll: False
         else:
-            self.should_log = console_log_level
+            # set should_log to a Callable[[State, LogLevel], bool]
+            if isinstance(console_log_level, LogLevel):
+                self.should_log = lambda state, ll: ll <= console_log_level
+            else:
+                self.should_log = console_log_level
+
+        # set the stream
         if isinstance(stream, str):
             if stream.lower() == 'stdout':
                 stream = sys.stdout
             elif stream.lower() == 'stderr':
                 stream = sys.stderr
             else:
-                raise ValueError(
-                    f"Invalid stream option: Should be 'stdout', 'stderr', or a TextIO-like object; got {stream}")
+                raise ValueError(f'stream must be one of ("stdout", "stderr", TextIO-like), got {stream}')
+
         self.stream = stream
 
+    @property
+    def current_pbar(self) -> Optional[_ProgressBar]:
+        if self.is_train is None:
+            return None
+        return self.train_pbar if self.is_train else self.eval_pbar
+
+    @current_pbar.setter
+    def current_pbar(self, pbar: Optional[_ProgressBar]):
+        if self.is_train is None:
```

Comment:
Since this is public, could this be a `ValueError` or a `RuntimeError`? Otherwise, if made private, could a standard `assert` statement be used?